src/Jackett.Server/Controllers/IndexerApiController.cs

```diff
@@ -167,11 +167,11 @@ private void ConfigureCacheResults(IEnumerable<TrackerCacheResult> results)
             var serverUrl = serverService.GetServerUrl(Request);
             foreach (var result in results)
             {
-                var link = result.Link;
                 var file = StringUtil.MakeValidFileName(result.Title, '_', false);
-                result.Link = serverService.ConvertToProxyLink(link, serverUrl, result.TrackerId, "dl", file);
+                result.Link = serverService.ConvertToProxyLink(result.Link, serverUrl, result.TrackerId, "dl", file);
+                result.Poster = serverService.ConvertToProxyLink(result.Poster, serverUrl, result.TrackerId, "img", "poster");
                 if (result.Link != null && result.Link.Scheme != "magnet" && !string.IsNullOrWhiteSpace(serverService.GetBlackholeDirectory()))
-                    result.BlackholeLink = serverService.ConvertToProxyLink(link, serverUrl, result.TrackerId, "bh", file);
+                    result.BlackholeLink = serverService.ConvertToProxyLink(result.Link, serverUrl, result.TrackerId, "bh", file);
             }
```

Comment:
looks like result.BlackholeLink is getting result.Link converted twice by serverService.ConvertToProxyLink, once at L171 and again at L174